tracer/src/Datadog.Trace.Tools.Runner/RunCiCommand.cs

```diff
@@ -108,6 +108,15 @@ public override async Task<int> ExecuteAsync(CommandContext context, RunCiSettin
 
                 // Upload git metadata by default (unless is disabled explicitly) or if ITR is enabled (required).
                 Log.Debug("RunCiCommand: Uploading repository changes.");
+
+                // Change the .git search folder to the CurrentDirectory or WorkingFolder
+                CIEnvironmentValues.Instance.GitSearchFolder = Environment.CurrentDirectory;
+                if (string.IsNullOrEmpty(CIEnvironmentValues.Instance.WorkspacePath))
```

Comment:
I don't really follow why you need this for local development? 🤔

Reply:
So as I was testing I realized that the runner was uploading the git metadata for `dd-trace-dotnet` and not the target app in my local, because the defaults uses the location of the runner assembly and not the current folder.

Reply:
oooooh, ok, I understand, thanks. It still makes me a little uneasy that we're adding code into the "real" runner to workaround an issue which is _presumably_ only specific to that one, local-testing/dev scenario? 🤔

Reply:
For me this is kind of a fix because the reason that the issue doesn't appear in CI testing is related to the CI environment variables. But if we execute the runner in an `unsupported` CI or in a docker container with the .git folder but not bypassing the env-vars the issue will be there. The Git information for the runner should be always extracted from the CurrentDirectory and not from the own runner folder, and this change does that. It's true I left the default behavior as a fallback and maybe I should remove that.

Reply:
I guess the thing I'm not really sure about is whether we should be using `WorkingDirectory`/`CurrentDirectory` at all🤔 Or should we be using the "target process" directory.

e.g. if you have something like this

```
.git
/something/.git
/something/subpath/.git
/something/subpath/MySampleApp.proj
```

Which of the 3 .git folders should we use in the following cases:

```bash
cd /something/subpath/ && dd-trace ci run -- dotnet test --project ./MySampleApp.proj
cd /something/ && dd-trace ci run -- dotnet test --project ./subpath/MySampleApp.proj
cd / && dd-trace ci run -- dotnet test --project ./something/subpath/MySampleApp.proj
```

(syntax may be off etc, but hopefully you get the idea)

Although, writing this out now, this is complicated further by the fact that `dotnet`'s path isn't the path you want, it's the project it references. I'm getting very confused now 😅

Reply:
yeah, I get it but for now we only support scenario 1: 

`cd /something/subpath/ && dd-trace ci run -- dotnet test --project ./MySampleApp.proj`

Here we will try to find the git folder starting from `/something/subpath/.git` -> `/something/.git` -> `/.git` 

The other 2 scenarios are harder because we don't parse the dotnet command at all, so we don't know the target project.

Keep in mind that this issue happens only on unsupported CI providers, for supported ones we extract everything from the CI environment variables.